### PR TITLE
CXX-896 Use chrono for TTL expiration options

### PR DESCRIPTION
**Pull request by acmorrow**

### CODE REVIEW

**Comment by acmorrow** [comment]
@xdg @hanumantmk @samantharitter Please take a look.

File: src/mongocxx/collection.cpp [hunk 1]
```diff
@@ -650,8 +650,11 @@ bsoncxx::document::value collection::create_index(view_or_value keys,
         }
     }
 
-    if (options.expire_after_seconds()) {
-        opt.expire_after_seconds = *options.expire_after_seconds();
+    if (options.expire_after()) {
+        const auto count = options.expire_after()->count();
+        if ((count < 0) || (count > std::numeric_limits<int32_t>::max()))
+            throw logic_error{error_code::k_invalid_parameter};;
```
Comment (xdg) [inline src/mongocxx/collection.cpp]: Double-semicolons.

**Comment by xdg** [comment]
And here's a test case: https://gist.github.com/xdg/f0ff32984218b0297df2dab989ec080b

**Comment by samantharitter** [comment]
LGTM mod David's semicolon note and additional test case

**Comment by Machyne** [comment]
Can we add an example or test showing how to set a ttl of 2 hours (e,g,) using `std::chrono::hours`?

**Comment by xdg** [comment]
@Machyne, you mean like `std::chrono::duration_cast<std::chrono::seconds>(std::chrono::hours(2))`?

I don't object, but I think it's clutter we don't really need.  Shouldn't we assume that users know how to convert durations (or can do the long-hand math from hours to seconds) on their own?

**Comment by acmorrow** [comment]
@Machyne @xdg They don't even need to do the conversion - it is baked in to std::chrono.

**Comment by Machyne** [comment]
@xdg @acmorrow - my thought was it would be nice to have an explicit example of something like:
```
options::index options{};
options.expire_after(std::chrono::hours{2});
coll.create_index(index_doc.view(), options);
```

File: src/mongocxx/test/collection_mocked.cpp [hunk 2]
```diff
@@ -272,7 +272,7 @@ TEST_CASE("Collection", "[collection]") {
         bool collection_create_index_called = false;
         bool success;
         bool expected_unique = true;
-        std::int32_t expected_expire_after_seconds = 500;
+        auto expected_expire_after = std::chrono::seconds(500);
```
Comment (Machyne) [inline src/mongocxx/test/collection_mocked.cpp]: Why not `std::chrono::seconds{500};`?

Comment (Machyne) [inline src/mongocxx/test/collection_mocked.cpp]: (with `{}` instead of `()`)

Comment (acmorrow) [inline src/mongocxx/test/collection_mocked.cpp]: Use of {} vs () is more a matter of style in cases like this. I don't feel a need to change it.

File: src/mongocxx/collection.cpp (hunk 1 above)
Comment (acmorrow) [inline src/mongocxx/collection.cpp]: Thanks, fixed

**Comment by acmorrow** [comment]
Updated. I merged in the test case from @xdg (thanks!), removed the double semi, and fixed up a minor formatting nit in the new test cases that clang-format pointed out (thanks clang-format!). PTAL.

File: src/mongocxx/test/collection_mocked.cpp [hunk 3]
```diff
@@ -38,7 +38,7 @@
 #include <mongocxx/read_preference.hpp>
 #include <mongocxx/stdx.hpp>
 #include <mongocxx/options/update.hpp>
-#include <mongocxx/exception/operation_exception.hpp>
+#include <mongocxx/exception/logic_error.hpp>
```
Comment (xdg) [inline src/mongocxx/test/collection_mocked.cpp]: For clarity: this is really two changes.  operation_exception was already imported above and thus removed here, and I added logic_error.

**Comment by xdg** [comment]
lgtm

File: src/mongocxx/test/collection_mocked.cpp (hunk 3 above)
Comment (acmorrow) [inline src/mongocxx/test/collection_mocked.cpp]: Oh, actually, thanks for pointing that out. This should be sorted in with the other includes. I will fix that before pushing.

**Pull request merged**